### ISSUE TITLE
REST API: X-On-Behalf-Of header doesn't seem to work

### ISSUE THREAD

**Issue by lpar** [zammad/zammad#1875]
### Infos:

* Used Zammad version: zammad-2.3.0-1518000657.f2e54de7.centos7.x86_64
* Installation method (source, package, ..): RPM
* Operating system: CentOS 7
* Database + version: postgresql-server-9.2.23-3.el7_4.x86_64
* Elasticsearch version: elasticsearch-5.6.7-1.noarch
* Browser + version: curl 7.55.1

### Expected behavior:

* [According to the docs](https://docs.zammad.org/en/latest/api-intro.html#example-curl-request-on-behalf-of-a-different-user) it should be possible to create a new ticket on behalf of a different existing user, so long as the API login user has admin privileges.

### Actual behavior:

* The ticket is created as owned by the API login user.

### Steps to reproduce the behavior:

```
curl --insecure -u "apiuser:apipassword" -H "Content-Type: application/json" -X POST 
  -d '{"firstname": "William", "lastname": "Parcher", "email": "parcher@mailinator.com", 
  "roles":["Customer"]}' https://zammad.example.com/api/v1/users
```

Results in output

```
{
  "id": 15,
  "organization_id": null,
  "login": "parcher@mailinator.com",
  "firstname": "William",
  "lastname": "Parcher",
  "email": "parcher@mailinator.com",
  "image": null,
  "image_source": null,
  "web": "",
  "phone": "",
  "fax": "",
  "mobile": "",
  "department": "",
  "street": "",
  "zip": "",
  "city": "",
  "country": "",
  "address": "",
  "vip": false,
  "verified": false,
  "active": true,
  "note": "",
  "last_login": null,
  "source": null,
  "login_failed": 0,
  "preferences": {
    
  },
  "updated_by_id": 14,
  "created_by_id": 14,
  "created_at": "2018-03-15T19:07:07.914Z",
  "updated_at": "2018-03-15T19:07:07.914Z",
  "out_of_office": false,
  "out_of_office_start_at": null,
  "out_of_office_end_at": null,
  "out_of_office_replacement_id": null,
  "role_ids": [
    3
  ],
  "organization_ids": [
    
  ],
  "authorization_ids": [
    
  ],
  "group_ids": {
    
  }
}
```

Then,

```
curl --insecure -u "apiuser:apipassword" -H "Content-Type: application/json" 
  -H "X-On-Behalf-Of: parcher@mailinator.com" -X POST 
  -d '{"title":"Test ticket title","group": "Groupname","article":{"subject":"Test ticket subject",
  "body":"This is the body of the message","type":"note","internal":false},
  "customer":"parcher@mailinator.com","note": "Note goes here"}' 
  https://zammad.example.com/api/v1/tickets
```

results in

```
{
  "id": 20,
  "group_id": 2,
  "priority_id": 2,
  "state_id": 1,
  "organization_id": null,
  "number": "23203",
  "title": "Test ticket title",
  "owner_id": 1,
  "customer_id": 14,
  "note": "Note goes here",
  "first_response_at": null,
  "first_response_escalation_at": null,
  "first_response_in_min": null,
  "first_response_diff_in_min": null,
  "close_at": null,
  "close_escalation_at": null,
  "close_in_min": null,
  "close_diff_in_min": null,
  "update_escalation_at": null,
  "update_in_min": null,
  "update_diff_in_min": null,
  "last_contact_at": null,
  "last_contact_agent_at": null,
  "last_contact_customer_at": null,
  "last_owner_update_at": null,
  "create_article_type_id": 10,
  "create_article_sender_id": 2,
  "article_count": 1,
  "escalation_at": null,
  "pending_time": null,
  "type": null,
  "time_unit": null,
  "preferences": {
    
  },
  "updated_by_id": 14,
  "created_by_id": 14,
  "created_at": "2018-03-15T19:13:48.550Z",
  "updated_at": "2018-03-15T19:13:48.625Z",
  "article_ids": [
    43
  ],
  "ticket_time_accounting_ids": [
    
  ]
}
```

Ticket is created successfully, but doesn't show as belonging to the created user. Instead, the apiuser is shown as the creator of the ticket. This matches the observed JSON returned, which shows the customer_id as 14 (the user number of the API user) rather than 15 (the user number of the user created by the first curl command).

**Comment by martini** [comment]
Hi @lpar 

[...]
https://docs.zammad.org/en/latest/api-intro.html#example-curl-request-on-behalf-of-a-different-user

Requirements for the feature:

* Authenticated user must have admin.user permissions
* Feature is available since Zammad version 2.4
[...]

Zammad 2.4 will come soon! :)

 -Martin

**Comment by lpar** [comment]
Oh dear. The word "since" led me to believe it was already out, and my eyes skimmed over the specific version number. Will try again when 2.4 is out.
